connector-drawer/src/test/java/com/redhat/cloud/notifications/connector/drawer/DrawerConnectorRoutesTest.java

```diff
@@ -0,0 +1,264 @@
+
+        JsonObject incomingPayload = buildIncomingPayload();
+
+        String cloudEventId = sendMessageToKafkaSource(incomingPayload);
+
+        JsonObject outcomingPayload = assertKafkaSinkIsSatisfied(cloudEventId, kafkaSinkMockEndpoint, false, null, RECIPIENTS_RESOLVER_EXCEPTION_MESSAGE);
+
+        checkRouteMetrics(ENGINE_TO_CONNECTOR, 1, 1, 1);
+        if (isConnectorRouteFailureHandled()) {
+            checkRouteMetrics(connectorConfig.getConnectorName(), 1, 1, 1);
+        } else {
+            checkRouteMetrics(connectorConfig.getConnectorName(), 0, 0, 1);
+        }
+        checkRouteMetrics(SUCCESS, 0, 0, 0);
+        checkRouteMetrics(CONNECTOR_TO_ENGINE, 0, 1, 1);
+        micrometerAssertionHelper.assertCounterIncrement(connectorConfig.getRedeliveryCounterName(), 0);
+        return outcomingPayload;
+    }
+
+    private JsonObject testSuccessfulDrawerNotification(int expectedMessagesOnDrawerTopic) throws Exception {
+
+        mockKafkaSourceEndpoint(); // This is the entry point of the connector.
+        MockEndpoint kafkaSinkMockEndpointDrawer = mockKafkaSinkEndpointDrawer(expectedMessagesOnDrawerTopic); // This is where the return message to the engine is sent.
+        MockEndpoint kafkaSinkMockEndpoint = mockKafkaSinkEndpoint(); // This is where the return message to the engine is sent.
+
+        JsonObject incomingPayload = buildIncomingPayload();
+
+        String cloudEventId = sendMessageToKafkaSource(incomingPayload);
+
+        JsonObject jsonObject = assertKafkaSinkIsSatisfied(cloudEventId, kafkaSinkMockEndpoint, true, null, "Event " + cloudEventId + " sent successfully");
+        assertKafkaSinkDrawerIsSatisfied(kafkaSinkMockEndpointDrawer);
+
+        checkRouteMetrics(ENGINE_TO_CONNECTOR, 0, 1, 1);
+        checkRouteMetrics(connectorConfig.getConnectorName(), 0, 1, 1);
+        checkRouteMetrics(SUCCESS, 0, 1, 1);
+        checkRouteMetrics(CONNECTOR_TO_ENGINE, 0, 1, 1);
+        micrometerAssertionHelper.assertCounterIncrement(connectorConfig.getRedeliveryCounterName(), 0);
+        return jsonObject;
+    }
+
+    protected MockEndpoint mockKafkaSinkEndpointDrawer(int expectedMessagesOnDrawerTopic) throws Exception {
+        adviceWith(CONNECTOR_TO_DRAWER, context(), new AdviceWithRouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                mockEndpointsAndSkip("kafka:" + drawerConnectorConfig.getOutgoingDrawerTopic());
+            }
+        });
+
+        MockEndpoint kafkaEndpoint = getMockEndpoint("mock:kafka:" + drawerConnectorConfig.getOutgoingDrawerTopic());
+        kafkaEndpoint.expectedMessageCount(expectedMessagesOnDrawerTopic);
+        return kafkaEndpoint;
+    }
+
+    protected static void assertKafkaSinkDrawerIsSatisfied(MockEndpoint kafkaSinkMockDrawerEndpoint) throws InterruptedException {
+
+        kafkaSinkMockDrawerEndpoint.assertIsSatisfied();
```

Comment:
```suggestion
        // We need a timeout here because SEDA processes the exchange from a different thread and a race condition may happen.
        kafkaSinkMockDrawerEndpoint.assertIsSatisfied(2000L);
```